jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/AbstractAdvanceRuntimeDataServiceImpl.java

```diff
@@ -55,58 +58,155 @@ public void setEmf(EntityManagerFactory emf) {
+                                                                                                           " INNER JOIN ProcessInstanceLog pil ON pil.processInstanceId = task.processInstanceId \n " +
+                                                                                                           derivedTables +
+                                                                                                           " WHERE  pil.processType = :processType " + globalWhere +
+                                                                                                           " ORDER BY task.id ASC ";
+
+        return queryProcessUserTasksByVariables(attributes, processVariables, taskVariables, owners, processType, varPrefix, queryContext, mainSQLProducer, this::collectUserTaskData);
+
+    }
+
+
+    protected <R> List<R> queryProcessUserTasksByVariables(List<QueryParam> attributesArg,
+                                                           List<QueryParam> processVariablesArg,
+                                                           List<QueryParam> taskVariablesArg,
+                                                           List<String> ownersArg,
+                                                           int processType,
+                                                           String varPrefix,
+                                                           QueryContext queryContext,
+                                                           BiFunction<StringBuilder, StringBuilder, String> mainSQLproducer,
+                                                           BiFunction<List<Number>, String, List<R>> dataCollector) {
+
+        List<QueryParam> attributes = attributesArg != null ? attributesArg : emptyList();
+        List<QueryParam> processVariables = processVariablesArg != null ? processVariablesArg : emptyList();
+        List<QueryParam> taskVariables = taskVariablesArg != null ? taskVariablesArg : emptyList();
+        List<String> owners = ownersArg != null ? ownersArg : emptyList();
+
+
+        StringBuilder globalWhere = new StringBuilder();
         StringBuilder derivedTables = new StringBuilder();
-        if (!variables.isEmpty()) {
-            List<String> conditions = new ArrayList<>();
-            variables.stream().forEach(expr -> conditions.add("(A1.variableId = :NAME_" + expr.getColumn() + " AND " + computeExpression(expr, "A1.value", ":VALUE_" + expr.getColumn()) + ")\n"));
-            String where = String.join(" OR ", conditions);
+        if (!taskVariables.isEmpty()) {
+            String where = computeVariableExpression(taskVariables, "V", "name", "value");
+            derivedTables.append("INNER JOIN (\n" +
+                                 "SELECT taskId \n" +
+                                 "FROM TaskVariableImpl \n" +
+                                 "WHERE type = 0 AND (" + where + ")\n" +
```

Comment:
I would use TaskVariable.VariableType.INPUT instead of hardcoding values in the query.
```suggestion
                                 "WHERE type = " + TaskVariable.VariableType.INPUT + " AND (" + where + ")\n" +
```